tools/ci_build/github/linux/ubuntu16.04/install.sh

```diff
@@ -27,7 +27,11 @@ apt-get update && apt-get install -y --no-install-recommends \
         bzip2 \
         unzip \
         rsync libunwind8 \
-        python3-setuptools python3-numpy python3-wheel python python3-pip
+        python3-setuptools python3-numpy python3-wheel python python3-pip \
+        software-properties-common
+
+add-apt-repository ppa:ubuntu-toolchain-r/test
+apt-get update && apt-get install -y --no-install-recommends g++-7
```

Comment:
We don't want to upgrade our compiler version.

Reply:
Ha yeah we know. The reason for this is vcpkg needs g++7. I believe g++7 will only be installed alongside the system default and will not be the compiler for the rest of the project. We talked to @pranavsharma about it and he said it's OK for now.

That said, vcpkg will be removed soon if you're wary.

Reply:
This changes means, people must install gcc-7 first, to run our prebuilt binaries. 
As gcc-7 is not part of the official repository, I don't think it's a good idea. 

If you change this script, it will impact all of our CI pipelines.  It's a bit risky.

Reply:
g++-7 has been removed but we still install g++ in this script. Is that OK?

Reply:
Thanks!

Reply:
I've also removed g++ from this script. Not needed.